core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -47,18 +66,23 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
     }
 }
 
+// is File a could type to use here?
+// No way to get script name if file is not present, should add another property (val fileName: String)
 interface ScriptContents {
     val file: File?
     val annotations: Iterable<Annotation>
     val text: CharSequence?
```

Comment:
Because `CharSequence` is more generic and should be enough for the most of the actual use cases - searching or parsing. But I'm not really against making it a `String`, provided that we have a reason.